lisa/energy_model.py

```diff
@@ -908,25 +909,19 @@ def from_simplifiedEM_target(cls, target,
             directory='/sys/devices/system/cpu/energy_model'):
         """
         Create an EnergyModel by reading a target filesystem on a device with
-        the new Simplified Energy Model present.
+        the new Simplified Energy Model present in sysfs.
 
-        This uses the energy_model sysctl added by EAS patches to exposes
-        the frequency domains, together with a tuple of capacity, frequency
-        and active power for each CPU. This feature is not upstream in mainline
-        Linux (as of v4.17), and only exists in Android kernels later than
-        android-4.14.
-
-        Wrt. idle states - the EnergyModel constructed won't be aware of
-        any power data or topological dependencies for entering "cluster"
-        idle states since the simplified model has no such concept.
-
-        Initialises only Active States for CPUs and clears all other levels.
+        The patches exposing the Energy Model in sysfs have been abandonned
+        and this way of loading it is now deprecated.
 
         :param target: Devlib target object to read filesystem from. Must have
                        cpufreq and cpuidle modules enabled.
         :returns: Constructed EnergyModel object based on the parameters
                   reported by the target.
         """
+        warning.warn('The Energy Model in sysfs is DEPRECATED. Please use '
+                      'debugfs instead.')
```

Comment:
`DeprecationWarning` can be specified as 2nd parameter to `warning.warn()` apparently:
https://docs.python.org/3/library/warnings.html#warnings.warn

Reply:
I was wondering if `logging.warn()` wouldn't be better here actually ? That'd be consistent with the other warnings. I don't mind either way.

Reply:
The warning module allows the end user to filter out or in the warnings, or turn them into exceptions (to cleanup from deprecation warning for example). The warning module should be setup to redirect by default to logging, so it should give roughly the same result. (maybe we will need to make sure the setup is done this way in setup_logging(), but we can look at that later).

IMO, warning module is the right tool for that kind of things.

Reply:
OK. It seems that by default `DeprecationWarning` is ignored. Is this something we'd like to change in our logging setup then ? I think I'd like users to see this message by default. This is probably for another PR, though.

https://docs.python.org/3/library/warnings.html#updating-code-for-new-versions-of-dependencies 

> Notably, this “ignored by default” list includes DeprecationWarning (for every module except \_\_main\_\_), which means developers should make sure to test their code with typically ignored warningsmade visible in order to receive timely notifications of future breaking API changes (whether in the standard library or third party packages).

Reply:
It is, I'll update our `setup_logging` function to redirect `warnings.warn` as `logging.warn` by default, so everybody sees them (and if they want, they can override that to turn it into an exception).